src/game/engine/display.cpp

```diff
@@ -1137,15 +1986,127 @@ cell_t DisplayClass::Set_Cursor_Pos(cell_t cell)
     return retval;
 }
 
+/**
+ * Helper function handling the proximity check part.
+ */
+void DisplayClass::Check_Proximity(ObjectTypeClass *object, HousesType house, cell_t cell, int &passes, bool &outside_radar) const
+{
+    BuildingTypeClass *bld = reinterpret_cast<BuildingTypeClass *>(object);
+
+    if (!In_Radar(cell)) {
+        outside_radar = true;
+        passes = 0;
+        return;
+    }
+
+    // check adjacent cells in all directions until we meet the condition to stop
+    for (FacingType i = FACING_NORTH; i < FACING_COUNT; ++i) {
+        cell_t adj_cell = Cell_Get_Adjacent(cell, i);
+        if (In_Radar(adj_cell)) {
+            // is the adjacent cell a Wall or Bib owned by this house?
+            if (bld->Is_Wall() || Array[adj_cell].Has_Bib()) {
+                if (Array[adj_cell].Owner() == house) {
+                    passes = 1;
+                    return;
+                }
+            }
+
+            // is the adjacent cell a building owned by this house and is BaseNormal?
+            BuildingClass *tptr = (BuildingClass *)Array[adj_cell].Cell_Techno(0, 0);
+            if (tptr != nullptr && tptr->What_Am_I() == RTTI_BUILDING && tptr->Get_Owner_House()->What_Type() == house) {
+                if (tptr->Class_Of().Base_Normal()) {
+                    passes = 1;
+                    return;
+                }
+            }
+
+            // proximity hasn't passed yet so check all neighboring cells in all directions until we meet the condition to
+            // stop
+            for (FacingType j = FACING_NORTH; j < FACING_COUNT; ++j) {
+                cell_t next_cell = Cell_Get_Adjacent(adj_cell, j);
+                // is the cell next to the adjacent cell a Wall or Bib owned by this house?
+                if (bld->Is_Wall() || Array[next_cell].Has_Bib()) {
```

Comment:
We repeat Array[next_cell] here, is it worth fetching the CellClass instance once and using this for both checks?

Reply:
I'm inclined to leave this for now as the original code appears to work the same way and compilers might even optimise this now.